nixos/modules/programs/steam.nix

```diff
@@ -172,6 +185,8 @@ in {
       };
     };
 
+    programs.steam.extraPackages = cfg.fontPackages;
```

Comment:
Isn't this a breaking change?

Reply:
Why do you think so?

Reply:
It should fix the CJK font display on Steam and introduce no other side effects. Whether to treat it as a breaking or fix commit depends on you. I just open it to discuss.

Reply:
> Why do you think so?

Previously, `extraPackages` defaulted to `[ ]`.

https://github.com/NixOS/nixpkgs/blob/daa52e1d9e89a4df5b844d26225db10fc9475739/nixos/modules/programs/steam.nix#L75-L77

And now you have `programs.steam.extraPackages = cfg.fontPackages;` in addition. It's not breaking if `fontPackages` also defaults to `[ ]`, but it effectively has the value of `config.fonts.packages` instead. So, 

> It should fix the CJK font display on Steam and introduce no other side effects.

This will not fix the display if no CJK font is installed system-wide. But in addition, there is the side effect of having all fonts passed via `extraPkgs` to `steamPackages.steam-fhsenv` on every installation with Steam enabled.

Reply:
I still cannot see why the default non-empty `cfg.fontPackages` is a breaking change.  Since the type of `programs.steam.extraPackages` is a list, the value of `programs.steam.extraPackages` is the merge (concat) result of newly added `cfg.fontPackages` and users' config.

Reply:
Ah, I see.  This may change the font steam uses.  If the previous steam font is fine for users, this can be seen as a breaking change.

Reply:
Feel free to close it if I'm wrong